### ISSUE TITLE
Few samples

### ISSUE THREAD

**Issue by salmonfish9** [zou4all/EcoAI-Intelligent-Innovations-for-a-Greener-Future#1]
Hi @zou4all,
Thanks for sharing this!
Does this work if I have just few samples of the object to be detected on Satellite images? That means I don't have hundereds of samples to train the model with it.

**Comment by zou4all** [comment]
Hello @salmonfish9, 
Thank you for your comment. I hope you enjoy this preliminary study :)
YOLO can performs well  with small datasets thanks to pretrained weights, the efficient architecture and the strong data augmentation.
I have already worked on my initial attempts with another small dataset, but I modified some training parameters, such as increasing the number of epochs..
Here is my example along with the corresponding results of the model : 
 https://github.com/zou4all/EcoAI-Intelligent-Innovations-for-a-Greener-Future/tree/main/EcoAI-CaseStudy/EcoAI-V1.1

**Comment by salmonfish9** [comment]
Hi @zou4all ,
Thanks for your reply :)

By few samples I meant less than 5 samples, does this work?

Also, are these 24 samples you have used in the example you have mentioned above before augmentation or after it?

**Comment by zou4all** [comment]
Hello again! 😊
Hello, the 24 samples are original images without any augmentation. However, here's a similar scenario where data augmentation was applied to a smaller dataset before starting the training process.
https://github.com/zou4all/EcoAI-Intelligent-Innovations-for-a-Greener-Future/tree/main/EcoAI-CaseStudy/EcoAI-V1.3

**Comment by salmonfish9** [comment]
Greeting @zou4all,
Thanks again for sharing this!
I have a question regarding Data Augmentation. That's should I label the augmented images manually, or are there an automated method of doing that? I have heared that YOLO has a built-in augmentation process that does the agumentation (along with modifying the labels accordingly) on the fly.

**Comment by zou4all** [comment]
Oh, its a very good question @salmonfish9 ^^
With only 5 images, it is essential to perform offline augmentation before training in addition to using YOLOv5’s built-in augmentation to prevent overfitting. In fact, with such a small dataset, the model is likely to memorize the training data instead of generalizing to new images.

And to ensure that the bounding boxes remain the labels in the augmented images, 
we can use the BboxParams function from Albumentations, as demonstrated in this script :
https://github.com/zou4all/EcoAI-Intelligent-Innovations-for-a-Greener-Future/blob/main/EcoAI-CaseStudy/EcoAI-V1.3/Data-Augm/data_aug.py

Once the dataset has been augmented, we can further enhance the training process by enabling YOLOv5’s built-in augmentation using this command:
python train.py --img 640 --batch 16 --epochs 50 --data data.yaml --weights yolov5s.pt --hyp data/hyp.scratch.yaml
(The hyp.scratch.yaml file automatically applies augmentations)

**Comment by salmonfish9** [comment]
Thanks a lot @zou4all, that makes sense.
